### PR TITLE
WINDUP-1677 inconsistent type value annotations

### PR DESCRIPTION
**Pull request by klinki**

### CODE REVIEW

File: graph/api/src/main/java/org/jboss/windup/graph/model/ArchiveModel.java [hunk 1]
```diff
@@ -19,7 +19,7 @@
 @TypeValue(ArchiveModel.TYPE)
 public interface ArchiveModel extends FileModel
 {
-    String TYPE = "ArchiveModel:";
+    String TYPE = "ArchiveModel";
     String ARCHIVE_NAME = TYPE + "archiveName";
```
Comment (mareknovotny) [inline graph/api/src/main/java/org/jboss/windup/graph/model/ArchiveModel.java]: you should have here colon you deleted in TYPE

**Comment by mareknovotny** [comment]
the result is not still consistent, somewhere you removed colons from TYPE and they are missing in concatenated strings for complete properties.
The other things is that somewhere is used colons and somewhere dash.

File: graph/api/src/main/java/org/jboss/windup/graph/model/ApplicationArchiveModel.java [hunk 2]
```diff
@@ -2,7 +2,8 @@
 
 import com.tinkerpop.frames.modules.typedgraph.TypeValue;
 
-@TypeValue("ApplicationArchive")
+@TypeValue(ApplicationArchiveModel.TYPE)
 public interface ApplicationArchiveModel extends ApplicationModel, ArchiveModel
 {
+    String TYPE = "ApplicationArchiveModel";
```
Comment (OndraZizka) [inline graph/api/src/main/java/org/jboss/windup/graph/model/ApplicationArchiveModel.java]: I would generally remove the Model suffix, it's not really needed and makes everything longer - esp. the JSON data, URLs etc.

File: graph/api/src/main/java/org/jboss/windup/graph/model/ArchiveModel.java (hunk 1 above)
Comment (OndraZizka) [reply graph/api/src/main/java/org/jboss/windup/graph/model/ArchiveModel.java]: The colon just delimits the type discriminator from edge label. Could be -, which makes less troubles with indexes IIRC.

File: reporting/api/src/main/java/org/jboss/windup/reporting/model/InlineHintModel.java [hunk 3]
```diff
@@ -17,7 +17,7 @@
 @TypeValue(InlineHintModel.TYPE)
 public interface InlineHintModel extends EffortReportModel, FileLocationModel, TaggableModel
 {
-    String TYPE = "Hint";
+    String TYPE = "InlineHintModel";
```
Comment (OndraZizka) [inline reporting/api/src/main/java/org/jboss/windup/reporting/model/InlineHintModel.java]: Same here. If there is 10000 hints, this adds 110 kb to the JSON data. Not sure if we care.

File: graph/api/src/main/java/org/jboss/windup/graph/model/ArchiveModel.java (hunk 1 above)
Comment (mareknovotny) [reply graph/api/src/main/java/org/jboss/windup/graph/model/ArchiveModel.java]: yep, dash or colon, if we prefer dash as it is more convenient for indexes, we should go with dashes everywhere, do you agree @OndraZizka ?

Comment (klinki) [reply graph/api/src/main/java/org/jboss/windup/graph/model/ArchiveModel.java]: I don't understand why `ARCHIVE_NAME` has `TYPE` prefix, but other properties (`UNZIPPED_DIRECTORY, PARENT_ARCHIVE` don't. 

This seems like another inconsistency. Should there be `TYPE` prefix or not?

Comment (mareknovotny) [reply graph/api/src/main/java/org/jboss/windup/graph/model/ArchiveModel.java]: That was the random thing to satisfy uniqueness, but here that goes as we have bunch of models, types, properties and checks for uniqueness needs to be done manually until somebody experience exceptions in runtime :(

Comment (jsight) [reply graph/api/src/main/java/org/jboss/windup/graph/model/ArchiveModel.java]: +1, personally, I find the prefixes to be a bit distracting.

File: graph/api/src/main/java/org/jboss/windup/graph/model/ApplicationArchiveModel.java (hunk 2 above)
Comment (jsight) [reply graph/api/src/main/java/org/jboss/windup/graph/model/ApplicationArchiveModel.java]: +1

**Pull request merged**